tests/test_aea.py

```diff
@@ -94,47 +82,41 @@ def test_initialise_aea():
 
 def test_act():
     """Tests the act function of the AEA."""
-    with LocalNode() as node:
-        agent_name = "MyAgent"
-        private_key_path = os.path.join(CUR_PATH, "data", "fet_private_key.txt")
-        wallet = Wallet({FETCHAI: private_key_path})
-        identity = Identity(agent_name, address=wallet.addresses[FETCHAI])
-        ledger_apis = LedgerApis({}, FETCHAI)
-        connections = [
-            OEFLocalConnection(
-                identity.address, node, connection_id=LOCAL_CONNECTION_PUBLIC_ID
-            )
-        ]
-        resources = Resources(str(Path(CUR_PATH, "data", "dummy_aea")))
-
-        agent = AEA(
-            identity, connections, wallet, ledger_apis, resources, is_programmatic=False
-        )
-        t = Thread(target=agent.start)
-        try:
-            t.start()
-            time.sleep(1.0)
+    agent_name = "MyAgent"
+    private_key_path = os.path.join(CUR_PATH, "data", "fet_private_key.txt")
+    builder = AEABuilder()
+    builder.set_name(agent_name)
+    builder.add_private_key(FETCHAI, private_key_path)
+    builder.add_skill(Path(CUR_PATH, "data", "dummy_skill"))
+    agent = builder.build()
+    t = Thread(target=agent.start)
+    try:
+        t.start()
+        time.sleep(1.0)
 
-            behaviour = agent.resources.get_behaviour(DUMMY_SKILL_PUBLIC_ID, "dummy")
-            assert behaviour.nb_act_called > 0, "Act() wasn't called"
-        finally:
-            agent.stop()
-            t.join()
+        behaviour = agent.resources.get_behaviour(DUMMY_SKILL_PUBLIC_ID, "dummy")
+        assert behaviour.nb_act_called > 0, "Act() wasn't called"
+    finally:
+        agent.stop()
+        t.join()
 
 
 def test_react():
     """Tests income messages."""
     with LocalNode() as node:
         agent_name = "MyAgent"
         private_key_path = os.path.join(CUR_PATH, "data", "fet_private_key.txt")
-        wallet = Wallet({FETCHAI: private_key_path})
-        identity = Identity(agent_name, address=wallet.addresses[FETCHAI])
-        ledger_apis = LedgerApis({}, FETCHAI)
-        connection = OEFLocalConnection(
-            identity.address, node, connection_id=LOCAL_CONNECTION_PUBLIC_ID
+        builder = AEABuilder()
+        builder.set_name(agent_name)
+        builder.add_private_key(FETCHAI, private_key_path)
+        builder.add_connection(
+            Path(ROOT_DIR, "packages", "fetchai", "connections", "local")
         )
-        connections = [connection]
-        resources = Resources(str(Path(CUR_PATH, "data", "dummy_aea")))
```

Comment:
So we cannot build an entire AEA from a path?

Reply:
yes, I just wanted to build it partially. But probably it's not a bad idea to build from the directory in this case.